src/layouts/EditHomepage/EditHomepage.jsx

```diff
@@ -843,6 +984,20 @@ const EditHomepage = ({ match }) => {
         return newSection
       })
 
+      // Modify announcements date time object
+      if (filteredFrontMatter.announcements) {
+        filteredFrontMatter.announcements.announcement_items = filteredFrontMatter.announcements.announcement_items.map(
+          (announcement) => {
+            return announcement.$set(
```

Comment:
is this actually intended? `$set` doesn't exist as a primitive. do also note that this will only ever appear on our backend + templates -> will there be a visual drift? (displayed dt format on template is different from preview)

if so, maybe this modification should take place inside `onFieldChange` to preserve parity

Reply:
Hey, you make sense, mutating it only at the component level as a neater solution here, thanks for this! 352dda83e9ad59e741de145591ada8630fdc7fa1